### PR TITLE
Put back test in CI

### PR DESCRIPTION
**Pull request by benoit-cty**
#### Technical changes

- Put back test in CI
- Fix coveralls config fot GitHub Actions
- Add a test for `create_data_frame_by_entity`
- Bump Actions and Python version to fix warnings

### CODE REVIEW

File: openfisca_survey_manager/tests/test_create_data_frame_by_entity.py [hunk 1]
```diff
@@ -1,18 +1,35 @@
...
+            period=period,
+            index=True
+            )
+        for entity, input_dataframe in data_frame_by_entity.items():
+            print(f"{entity} for {period}")
+            print(input_dataframe.columns)
+            if entity == "person":
+                self.assertIn("person_id", input_dataframe.columns.to_list())
+            if entity == "household":
+                self.assertIn("household_id", input_dataframe.columns.to_list())
```
Comment (benoit-cty) [inline openfisca_survey_manager/tests/test_create_data_frame_by_entity.py:35]: Bonjour @benjello c'est cette ligne que l'on voudrait faire fonctionner. Ou alors SurveyManager sait merger les deux tables sans avoir besoin d'id ?
Voici l'erreur que cela génère : https://github.com/openfisca/openfisca-survey-manager/actions/runs/3566299257/jobs/5992587424#step:6:80

Comment (benjello) [reply openfisca_survey_manager/tests/test_create_data_frame_by_entity.py:35]: Thanks @benoit-cty for this work. Just wondering could you suggest a way of running the tests locally ?
Using `make install` ends up with this error message :
```
ERROR openfisca_survey_manager/__init__.py - NameError: name 'is_in_ci' is not defined
ERROR openfisca_survey_manager/__init__.py - NameError: name 'is_in_ci' is not defined
ERROR openfisca_survey_manager/aggregates.py - NameError: name 'is_in_ci' is not defined
ERROR openfisca_survey_manager/aggregates.py - NameError: name 'is_in_ci' is not defined
...
```

Comment (benoit-cty) [reply openfisca_survey_manager/tests/test_create_data_frame_by_entity.py:35]: My bad, I fixed it in my last commit.
You could also run the CI mode locally with `CI=true make test`.

**Comment by benjello** [comment]
@benoit-cty : I added names to the entities dataframes indexes (and modified the test accordingly). Does it help ?

File: .github/workflows/workflow.yml [hunk 2]
```diff
@@ -12,17 +12,17 @@ jobs:
       fail-fast: true
       matrix:
         os: ["ubuntu-latest"]  # On peut ajouter "macos-latest" si besoin
-        python-version: ["3.7.9", "3.8.9", "3.9.9"]
+        python-version: ["3.7.15", "3.8.14", "3.9.15"]
```
Comment (sandcha) [inline .github/workflows/workflow.yml:15]: Fixes CI as tested for example by https://github.com/openfisca/openfisca-france/pull/1930

**Pull request merged**